File: app/src/main/java/it/liceoarzignano/bold/ui/recyclerview/RecyclerTouchListener.java
package it.liceoarzignano.bold.ui.recyclerview;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

public class RecyclerTouchListener implements RecyclerView.OnItemTouchListener {
    private final GestureDetector mDetector;
    private final RecyclerClickListener mListener;

    public RecyclerTouchListener(Context context, final RecyclerClickListener listener) {
        this.mListener = listener;

        mDetector = new GestureDetector(context, new GestureDetector.SimpleOnGestureListener() {
            @Override
            public boolean onSingleTapUp(MotionEvent event) {
                return true;
            }
        });
    }

    @Override
    public boolean onInterceptTouchEvent(RecyclerView view, MotionEvent event) {
        View child = view.findChildViewUnder(event.getX(), event.getY());
        if (child != null && mListener != null && mDetector.onTouchEvent(event)) {
            mListener.onClick(child, view.getChildAdapterPosition(child));
            return true;
        }
        return false;
    }

    @Override
    public void onTouchEvent(RecyclerView view, MotionEvent event) {
    }

    @Override
    public void onRequestDisallowInterceptTouchEvent(boolean shouldDisallow) {
    }
}

File: app/src/main/kotlin/it/liceoarzignano/bold/ui/recyclerview/RecyclerTouchListener.kt
package it.liceoarzignano.bold.ui.recyclerview

import android.content.Context
import android.view.GestureDetector
import android.view.MotionEvent

class RecyclerTouchListener(context: Context, private val mListener: RecyclerClickListener?) :
        androidx.recyclerview.widget.RecyclerView.OnItemTouchListener {
    private val mDetector = GestureDetector(context,
            object : GestureDetector.SimpleOnGestureListener() {
                override fun onSingleTapUp(event: MotionEvent): Boolean = true
            })


    override fun onInterceptTouchEvent(view: androidx.recyclerview.widget.RecyclerView, event: MotionEvent): Boolean {
        val child = view.findChildViewUnder(event.x, event.y)
        if (child != null && mListener != null && mDetector.onTouchEvent(event)) {
            mListener.onClick(child, view.getChildAdapterPosition(child))
            return true
        }
        return false
    }

    override fun onTouchEvent(view: androidx.recyclerview.widget.RecyclerView, event: MotionEvent) = Unit

    override fun onRequestDisallowInterceptTouchEvent(shouldDisallow: Boolean) = Unit
}
